db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java

```diff
@@ -521,9 +521,7 @@ boolean cleanupOnError(Throwable e, boolean diagActive)
 			SanityManager.ASSERT(cm != null, "cannot cleanup on error with null context manager");
 
         //DERBY-4856 thread dump
-        boolean result = cm.cleanupOnError(e, diagActive);
-		csf.removeContext(cm);
-		return result;
+        return cm.cleanupOnError(e, diagActive);
```

Comment:
@dgomezferro You mentioned that csf.removeContext(cm) is called outside the function cleanupOnError() thus can be removed here, but I couldn't find it at the caller to cleanupOnError(), could you point me to the code?

Reply:
There are 3 call sites:
- TransactionResourceImpl.handleException()
- EmbedConnection.close()
- EmbedConnection constructor

The latter 2 close it explicitly calling restoreContextStack(), but the first one doesn't. Several callers of TransactionResourceImpl.handleException() close it too, but not all. At first this 'double' cleanup was creating problems for me, but now it seems to be fine, so I'll revert this change to keep the current behaviour.